amethyst_utils/src/ortho_camera.rs

```diff
@@ -0,0 +1,110 @@
+use amethyst_core::Axis2;
+use amethyst_core::cgmath::Ortho;
+use amethyst_core::specs::{Component, DenseVecStorage, Join, System, ReadExpect, ReadStorage, WriteStorage};
+use amethyst_renderer::{Camera, ScreenDimensions};
+
+/// `Component` attached to the camera's entity that allows automatically adjusting the camera's matrix according to preferences in the "mode" field.
+/// It tries as much as possible to ajust the camera so that the world's coordinate (0,0) is at the bottom left and (1,1) is at the top right of the window.
+#[derive(Default)]
+pub struct NormalOrthoCamera {
+    /// How the camera's matrix is changed when the window's aspect ratio changes. See `CameraNormalizeMode` for more info.
+    pub mode: CameraNormalizeMode,
+}
+
+impl NormalOrthoCamera {
+    pub fn camera_offsets(&self, ratio: f32) -> (f32,f32,f32,f32) {
+        self.mode.camera_offsets(ratio)
+    }
+}
+
+impl Component for NormalOrthoCamera {
+    type Storage = DenseVecStorage<Self>;
+}
+
+/// Settings that decide how to scale the camera's matrix when the aspect ratio changes.
+pub enum CameraNormalizeMode {
+    /// Using an aspect ratio of 1:1, tries to ajust the matrix values of the camera so
+    /// that the direction opposite to the stretch_direction always have a world size of 1.
+    /// If you use the default `Transform` (position = 0,0,0), this means that the direction opposite to stretch_direction 
+    /// will go from 0.0 to 1.0 in world coordinates.
+    /// Scene space can be lost on the specified stretch_direction.
+    Lossy {stretch_direction: Axis2},
+    
+    /// Scales the render dynamically to ensure no space is lost in the [0,1] range on any axis.
```

Comment:
Could you make it so that the actual [0;1] range be modifiable? Many workflow don't use [0,1].
An affine transformation should do the trick I think.

Reply:
You just made me realize I forgot to include all the unit test I made for this